internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -706,14 +769,14 @@ func flattenSqlVirtualMachineAutoPatching(autoPatching *sqlvirtualmachine.AutoPa
 		return []interface{}{}
 	}
 
-	var startHour int32
+	var startHour int
 	if autoPatching.MaintenanceWindowStartingHour != nil {
-		startHour = *autoPatching.MaintenanceWindowStartingHour
+		startHour = int(*autoPatching.MaintenanceWindowStartingHour)
```

Comment:
This is for prop validation. the prop retrieved from d.Get() is and interface{} of int, which will be different from int32